# Pregunta1
#### ¿Cuál opción corresponde al servicio de salud al que llegaron más pacientes el día 1?  



In [41]:
import pandas as pd

In [42]:
# Leer una hoja específica del archivo Excel
Hosp_sheet = pd.read_excel('Datos.xls', sheet_name='Hospitales')
Hosp_sheet

Nombre Servicio de Salud   
0   Metropolitano Norte                           ...  \
1   Metropolitano Occidente                       ...   
2   Metropolitano Occidente                       ...   
3   Metropolitano Occidente                       ...   
4   Metropolitano Occidente                       ...   
5   Metropolitano Central                         ...   
6   Metropolitano Central                         ...   
7   Metropolitano Central                         ...   
8   Metropolitano Oriente                         ...   
9   Metropolitano Oriente                         ...   
10  Metropolitano Oriente                         ...   
11  Metropolitano Oriente                         ...   
12  Metropolitano Sur                             ...   
13  Metropolitano Sur                             ...   
14  Metropolitano Sur                             ...   
15  Metropolitano Sur                             ...   
16  Metropolitano Sur                             ...   
17  Metropolitano Sur Oriente                     ...   
18  Metropolitano Sur Oriente                     ...   
19  Metropolitano Sur Oriente                     ...   
20  Metropolitano Sur Oriente                     ...   

                               Nombre Establecimiento   
0                      Complejo Hospitalario San José  \
1              Hospital Adalberto Steeger (Talagante)   
2                     Hospital Dr. Félix Bulnes Cerda   
3                           Hospital San Juan de Dios   
4        Instituto Traumatológico Dr. Teodoro Gebauer   
5                 Hospital Clínico San Borja-Arriarán   
6   Hospital de Urgencia Asistencia Pública Dr. Al...   
7                                  Hospital el Carmen   
8                               Hospital Del Salvador   
9                          Hospital Dr. Luis Tisné B.   
10       Instituto de Neurocirugía Dr. Alfonso Asenjo   
11  Instituto Nacional de Enfermedades Respiratori...   
12                       Hospital Barros Luco Trudeau   
13  Hospital de Enfermedades Infecciosas Dr. Lucio...   
14              Hospital Dr. Exequiel González Cortés   
15                                   Hospital El Pino   
16          Hospital Parroquial de San Bernardo         
17         Complejo Hospitalario Dr. Sótero del Río     
18                     Hospital Clínico de la Florida   
19              Hospital Metropolitano ( ex Militar )   
20         Hospital Padre Alberto Hurtado (San Ramón)   

    Área Cuidados Intensivos Adultos  Área Cuidados Intermedios Adulto   
0                                 11                                33  \
1                                  6                                11   
2                                  3                                 6   
3                                 29                                47   
4                                 12                                 6   
5                                 17                                26   
6                                 24                                20   
7                                  6                                12   
8                                 22                                24   
9                                  8                                16   
10                                12                                11   
11                                15                                29   
12                                31                                43   
13                                11                                 9   
14                                23                                 8   
15                                12                                20   
16                                 6                                10   
17                                20                                35   
18                                 9                                15   
19                                12   

In [43]:
HospitalxServ = Hosp_sheet[["Nombre Servicio de Salud","Nombre Establecimiento"]].groupby("Nombre Servicio de Salud") 
#de aqui saco las columnas que necesito para luego agruparlas
HxS = HospitalxServ.apply(lambda x: x).reset_index(drop=True)

In [44]:
Info_sheet = pd.read_excel('Datos.xls', sheet_name='Información Histórica') #Datos del dia de llegada
Info_sheet

Número de llegada                              Hospital al que llega   
0                      1       Instituto Traumatológico Dr. Teodoro Gebauer  \
1                      2                Hospital Clínico San Borja-Arriarán   
2                      3                         Hospital Dr. Luis Tisné B.   
3                      4              Hospital Dr. Exequiel González Cortés   
4                      5                       Hospital Barros Luco Trudeau   
...                  ...                                                ...   
65529              65530                Hospital Clínico San Borja-Arriarán   
65530              65531              Hospital Dr. Exequiel González Cortés   
65531              65532                              Hospital Del Salvador   
65532              65533  Hospital de Enfermedades Infecciosas Dr. Lucio...   
65533              65534  Hospital de Urgencia Asistencia Pública Dr. Al...   

         GRD  Día de llegada  Día de salida  Estadia  
0       GRD3               1              4        3  
1       GRD7               1              5        4  
2       GRD7               1              5        4  
3      GRD10               1             11       10  
4       GRD6               1              6        5  
...      ...             ...            ...      ...  
65529   GRD2             424            428        4  
65530   GRD9             424            429        5  
65531  GRD10             424            434       10  
65532   GRD3             424            427        3  
65533   GRD7             424            428        4  

[65534 rows x 6 columns]

In [45]:
df1 = Info_sheet.iloc[:146] #Recorto solo los del primer dia
#agrupo y cuento
result = df1.groupby(['Día de llegada', 'Hospital al que llega']).size().reset_index(name='Num_Pacientes') 
result.sort_values(by=["Num_Pacientes"],ascending = False) #Ordeno de mayor a menor


Día de llegada                              Hospital al que llega   
1                1                     Complejo Hospitalario San José  \
3                1                       Hospital Barros Luco Trudeau   
4                1                Hospital Clínico San Borja-Arriarán   
9                1                         Hospital Dr. Luis Tisné B.   
19               1       Instituto Traumatológico Dr. Teodoro Gebauer   
6                1                              Hospital Del Salvador   
0                1         Complejo Hospitalario Dr. Sótero del Río     
8                1                    Hospital Dr. Félix Bulnes Cerda   
16               1  Hospital de Urgencia Asistencia Pública Dr. Al...   
18               1  Instituto Nacional de Enfermedades Respiratori...   
7                1              Hospital Dr. Exequiel González Cortés   
20               1       Instituto de Neurocirugía Dr. Alfonso Asenjo   
14               1                          Hospital San Juan de Dios   
2                1             Hospital Adalberto Steeger (Talagante)   
12               1         Hospital Padre Alberto Hurtado (San Ramón)   
13               1          Hospital Parroquial de San Bernardo         
17               1                                 Hospital el Carmen   
10               1                                   Hospital El Pino   
15               1  Hospital de Enfermedades Infecciosas Dr. Lucio...   
11               1              Hospital Metropolitano ( ex Militar )   
5                1                     Hospital Clínico de la Florida   

    Num_Pacientes  
1              16  
3              14  
4              14  
9              13  
19             11  
6              11  
0              10  
8              10  
16              9  
18              7  
7               7  
20              5  
14              4  
2               4  
12              2  
13              2  
17              2  
10              2  
15              1  
11              1  
5               1

In [46]:
result = result.rename(columns={'Hospital al que llega' : 'Nombre Establecimiento'}) #primero modifico el nombre de las columnas para que se pueda hacer un merge
result

Día de llegada                             Nombre Establecimiento   
0                1         Complejo Hospitalario Dr. Sótero del Río    \
1                1                     Complejo Hospitalario San José   
2                1             Hospital Adalberto Steeger (Talagante)   
3                1                       Hospital Barros Luco Trudeau   
4                1                Hospital Clínico San Borja-Arriarán   
5                1                     Hospital Clínico de la Florida   
6                1                              Hospital Del Salvador   
7                1              Hospital Dr. Exequiel González Cortés   
8                1                    Hospital Dr. Félix Bulnes Cerda   
9                1                         Hospital Dr. Luis Tisné B.   
10               1                                   Hospital El Pino   
11               1              Hospital Metropolitano ( ex Militar )   
12               1         Hospital Padre Alberto Hurtado (San Ramón)   
13               1          Hospital Parroquial de San Bernardo         
14               1                          Hospital San Juan de Dios   
15               1  Hospital de Enfermedades Infecciosas Dr. Lucio...   
16               1  Hospital de Urgencia Asistencia Pública Dr. Al...   
17               1                                 Hospital el Carmen   
18               1  Instituto Nacional de Enfermedades Respiratori...   
19               1       Instituto Traumatológico Dr. Teodoro Gebauer   
20               1       Instituto de Neurocirugía Dr. Alfonso Asenjo   

    Num_Pacientes  
0              10  
1              16  
2               4  
3              14  
4              14  
5               1  
6              11  
7               7  
8              10  
9              13  
10              2  
11              1  
12              2  
13              2  
14              4  
15              1  
16              9  
17              2  
18              7  
19             11  
20              5

In [47]:
df_combinado = pd.merge(result, HxS, on='Nombre Establecimiento') #combino ambos df por la columna comun
df_combinado #DF combinado

Día de llegada                             Nombre Establecimiento   
0                1         Complejo Hospitalario Dr. Sótero del Río    \
1                1                     Complejo Hospitalario San José   
2                1             Hospital Adalberto Steeger (Talagante)   
3                1                       Hospital Barros Luco Trudeau   
4                1                Hospital Clínico San Borja-Arriarán   
5                1                     Hospital Clínico de la Florida   
6                1                              Hospital Del Salvador   
7                1              Hospital Dr. Exequiel González Cortés   
8                1                    Hospital Dr. Félix Bulnes Cerda   
9                1                         Hospital Dr. Luis Tisné B.   
10               1                                   Hospital El Pino   
11               1              Hospital Metropolitano ( ex Militar )   
12               1         Hospital Padre Alberto Hurtado (San Ramón)   
13               1          Hospital Parroquial de San Bernardo         
14               1                          Hospital San Juan de Dios   
15               1  Hospital de Enfermedades Infecciosas Dr. Lucio...   
16               1  Hospital de Urgencia Asistencia Pública Dr. Al...   
17               1                                 Hospital el Carmen   
18               1  Instituto Nacional de Enfermedades Respiratori...   
19               1       Instituto Traumatológico Dr. Teodoro Gebauer   
20               1       Instituto de Neurocirugía Dr. Alfonso Asenjo   

    Num_Pacientes                           Nombre Servicio de Salud  
0              10  Metropolitano Sur Oriente                     ...  
1              16  Metropolitano Norte                           ...  
2               4  Metropolitano Occidente                       ...  
3              14  Metropolitano Sur                             ...  
4              14  Metropolitano Central                         ...  
5               1  Metropolitano Sur Oriente                     ...  
6              11  Metropolitano Oriente                         ...  
7               7  Metropolitano Sur                             ...  
8              10  Metropolitano Occidente                       ...  
9              13  Metropolitano Oriente                         ...  
10              2  Metropolitano Sur                             ...  
11              1  Metropolitano Sur Oriente                     ...  
12              2  Metropolitano Sur Oriente                     ...  
13              2  Metropolitano Sur                             ...  
14              4  Metropolitano Occidente                       ...  
15              1  Metropolitano Sur                             ...  
16              9  Metropolitano Central                         ...  
17              2  Metropolitano Central                         ...  
18              7  Metropolitano Oriente                         ...  
19             11  Metropolitano Occidente                       ...  
20              5  Metropolitano Oriente                         ...

In [48]:
result1 = df_combinado.groupby('Nombre Servicio de Salud')['Num_Pacientes'].sum().reset_index(name='Total_Pacientes')#Por ultimo agrupo y sumo los pacientes por servicio de salud
result1

Nombre Servicio de Salud  Total_Pacientes
0  Metropolitano Central                         ...               25
1  Metropolitano Norte                           ...               16
2  Metropolitano Occidente                       ...               29
3  Metropolitano Oriente                         ...               36
4  Metropolitano Sur                             ...               26
5  Metropolitano Sur Oriente                     ...               14

### Entonces el servicio de salud al que llegaron mas pacientes fue el Metropolitano Central 

# Pregunta 2
### ¿Cuál cifra corresponde a la cantidad de pacientes de cada GRD 1, 7 y 10 y que han llegado históricamente al servicio de salud Metropolitano Sur?  



In [49]:
Info_sheet = pd.read_excel('Datos.xls', sheet_name='Información Histórica') #Datos del dia de llegada
Info_sheet

Número de llegada                              Hospital al que llega   
0                      1       Instituto Traumatológico Dr. Teodoro Gebauer  \
1                      2                Hospital Clínico San Borja-Arriarán   
2                      3                         Hospital Dr. Luis Tisné B.   
3                      4              Hospital Dr. Exequiel González Cortés   
4                      5                       Hospital Barros Luco Trudeau   
...                  ...                                                ...   
65529              65530                Hospital Clínico San Borja-Arriarán   
65530              65531              Hospital Dr. Exequiel González Cortés   
65531              65532                              Hospital Del Salvador   
65532              65533  Hospital de Enfermedades Infecciosas Dr. Lucio...   
65533              65534  Hospital de Urgencia Asistencia Pública Dr. Al...   

         GRD  Día de llegada  Día de salida  Estadia  
0       GRD3               1              4        3  
1       GRD7               1              5        4  
2       GRD7               1              5        4  
3      GRD10               1             11       10  
4       GRD6               1              6        5  
...      ...             ...            ...      ...  
65529   GRD2             424            428        4  
65530   GRD9             424            429        5  
65531  GRD10             424            434       10  
65532   GRD3             424            427        3  
65533   GRD7             424            428        4  

[65534 rows x 6 columns]

In [50]:
result2 = Info_sheet.groupby(['Día de llegada', 'Hospital al que llega',"GRD"]).size().reset_index(name='Num_Pacientes') 
result2.sort_values(by=["Num_Pacientes"],ascending = False) #Ordeno de mayor a menor
result2 = result2.rename(columns={'Hospital al que llega' : 'Nombre Establecimiento'})
result2

Día de llegada                        Nombre Establecimiento   GRD   
0                   1    Complejo Hospitalario Dr. Sótero del Río    GRD1  \
1                   1    Complejo Hospitalario Dr. Sótero del Río    GRD2   
2                   1    Complejo Hospitalario Dr. Sótero del Río    GRD3   
3                   1    Complejo Hospitalario Dr. Sótero del Río    GRD4   
4                   1    Complejo Hospitalario Dr. Sótero del Río    GRD5   
...               ...                                           ...   ...   
44277             424  Instituto Traumatológico Dr. Teodoro Gebauer  GRD4   
44278             424  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD3   
44279             424  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD5   
44280             424  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD7   
44281             424  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD9   

       Num_Pacientes  
0                  1  
1                  1  
2                  1  
3                  1  
4                  2  
...              ...  
44277              1  
44278              2  
44279              2  
44280              1  
44281              2  

[44282 rows x 4 columns]

In [51]:
df_combinado2 = pd.merge(result2, HxS, on='Nombre Establecimiento')
df_combinado2

Día de llegada                        Nombre Establecimiento   GRD   
0                   1    Complejo Hospitalario Dr. Sótero del Río    GRD1  \
1                   1    Complejo Hospitalario Dr. Sótero del Río    GRD2   
2                   1    Complejo Hospitalario Dr. Sótero del Río    GRD3   
3                   1    Complejo Hospitalario Dr. Sótero del Río    GRD4   
4                   1    Complejo Hospitalario Dr. Sótero del Río    GRD5   
...               ...                                           ...   ...   
44277             423  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD9   
44278             424  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD3   
44279             424  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD5   
44280             424  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD7   
44281             424  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD9   

       Num_Pacientes                           Nombre Servicio de Salud  
0                  1  Metropolitano Sur Oriente                     ...  
1                  1  Metropolitano Sur Oriente                     ...  
2                  1  Metropolitano Sur Oriente                     ...  
3                  1  Metropolitano Sur Oriente                     ...  
4                  2  Metropolitano Sur Oriente                     ...  
...              ...                                                ...  
44277              1  Metropolitano Oriente                         ...  
44278              2  Metropolitano Oriente                         ...  
44279              2  Metropolitano Oriente                         ...  
44280              1  Metropolitano Oriente                         ...  
44281              2  Metropolitano Oriente                         ...  

[44282 rows x 5 columns]

In [52]:
result2 = df_combinado2.groupby(['Nombre Servicio de Salud',"GRD"])['Num_Pacientes'].sum().reset_index(name='Total_Pacientes')
result2

Nombre Servicio de Salud    GRD  Total_Pacientes
0   Metropolitano Central                         ...   GRD1             1095
1   Metropolitano Central                         ...  GRD10             1111
2   Metropolitano Central                         ...   GRD2             1138
3   Metropolitano Central                         ...   GRD3             1084
4   Metropolitano Central                         ...   GRD4             1076
5   Metropolitano Central                         ...   GRD5             1120
6   Metropolitano Central                         ...   GRD6             1125
7   Metropolitano Central                         ...   GRD7             1118
8   Metropolitano Central                         ...   GRD8             1096
9   Metropolitano Central                         ...   GRD9             1127
10  Metropolitano Norte                           ...   GRD1              464
11  Metropolitano Norte                           ...  GRD10              471
12  Metropolitano Norte                           ...   GRD2              451
13  Metropolitano Norte                           ...   GRD3              454
14  Metropolitano Norte                           ...   GRD4              466
15  Metropolitano Norte                           ...   GRD5              490
16  Metropolitano Norte                           ...   GRD6              443
17  Metropolitano Norte                           ...   GRD7              506
18  Metropolitano Norte                           ...   GRD8              427
19  Metropolitano Norte                           ...   GRD9              480
20  Metropolitano Occidente                       ...   GRD1             1159
21  Metropolitano Occidente                       ...  GRD10             1152
22  Metropolitano Occidente                       ...   GRD2             1123
23  Metropolitano Occidente                       ...   GRD3             1153
24  Metropolitano Occidente                       ...   GRD4             1109
25  Metropolitano Occidente                       ...   GRD5             1207
26  Metropolitano Occidente                       ...   GRD6             1108
27  Metropolitano Occidente                       ...   GRD7             1141
28  Metropolitano Occidente                       ...   GRD8             1152
29  Metropolitano Occidente                       ...   GRD9             1111
30  Metropolitano Oriente                         ...   GRD1             1622
31  Metropolitano Oriente                         ...  GRD10             1559
32  Metropolitano Oriente                         ...   GRD2             1591
33  Metropolitano Oriente                         ...   GRD3             1574
34  Metropolitano Oriente                         ...   GRD4             1603
35  Metropolitano Oriente                         ...   GRD5             1664
36  Metropolitano Oriente                         ...   GRD6             1604
37  Metropolitano Oriente                         ...   GRD7             1608
38  Metropolitano Oriente                         ...   GRD8             1665
39  Metropolitano Oriente                         ...   GRD9             1642
40  Metropolitano Sur                             ...   GRD1             1311
41  Metropolitano Sur                             ...  GRD10             1269
42  Metropolitano Sur                             ...   GRD2             1237
43  Metropolitano Sur                             ...   GRD3             1337
44  Metropolitano Sur                             ...   GRD4             1248
45  Metropolitano Sur                             ...   GRD5             1256
46  Metropolitano Sur                             ...   GRD6             1286
47  Metropolitano Sur                             ...   GRD7             1293
48  Metropolitano Sur                             ...   GRD8             1226
49  Metropolitano Sur                             ...   GRD9             1327
50  Metropolitano Sur Oriente                     .

### 1311, 1293 y 1269  Respectivamen

# Pregunta 3
### ¿Cuál cifra corresponde a la mediana del número de pacientes que llegaron a cada servicio de salud?  



### Primero necesito el numero de dias como indice


In [53]:
result3_1 = Info_sheet.groupby(['Día de llegada', 'Hospital al que llega']).size().reset_index(name='Num_Pacientes') 
result3_1
result3_1 = result3_1.rename(columns={'Hospital al que llega' : 'Nombre Establecimiento'})
df_combinado3 = pd.merge(result3_1, HxS, on='Nombre Establecimiento') #combino ambos df por la columna comun
df_combinado3.head()

Día de llegada                      Nombre Establecimiento  Num_Pacientes   
0               1  Complejo Hospitalario Dr. Sótero del Río               10  \
1               2  Complejo Hospitalario Dr. Sótero del Río               12   
2               3  Complejo Hospitalario Dr. Sótero del Río                7   
3               4  Complejo Hospitalario Dr. Sótero del Río               11   
4               5  Complejo Hospitalario Dr. Sótero del Río               15   

                            Nombre Servicio de Salud  
0  Metropolitano Sur Oriente                     ...  
1  Metropolitano Sur Oriente                     ...  
2  Metropolitano Sur Oriente                     ...  
3  Metropolitano Sur Oriente                     ...  
4  Metropolitano Sur Oriente                     ...

In [54]:
df_pivot= df_combinado3.pivot(columns='Nombre Servicio de Salud', values='Num_Pacientes')
df_pivot

Nombre Servicio de Salud  Metropolitano Central                                
0                                                                       NaN   \
1                                                                       NaN    
2                                                                       NaN    
3                                                                       NaN    
4                                                                       NaN    
...                                                                     ...    
8810                                                                    NaN    
8811                                                                    NaN    
8812                                                                    NaN    
8813                                                                    NaN    
8814                                                                    NaN    

Nombre Servicio de Salud  Metropolitano Norte                                  
0                                                                       NaN   \
1                                                                       NaN    
2                                                                       NaN    
3                                                                       NaN    
4                                                                       NaN    
...                                                                     ...    
8810                                                                    NaN    
8811                                                                    NaN    
8812                                                                    NaN    
8813                                                                    NaN    
8814                                                                    NaN    

Nombre Servicio de Salud  Metropolitano Occidente                              
0                                                                       NaN   \
1                                                                       NaN    
2                                                                       NaN    
3                                                                       NaN    
4                                                                       NaN    
...                                                                     ...    
8810                                                                    NaN    
8811                                                                    NaN    
8812                                                                    NaN    
8813                                                                    NaN    
8814                                                                    NaN    

Nombre Servicio de Salud  Metropolitano Oriente                                
0                                                                       NaN   \
1                                                                       NaN    
2                                                                       NaN    
3                                                                       NaN    
4                                                                       NaN    
...                                                                     ...    
8810                                                                    8.0    
8811                                                                    4.0    
8812                                                                    6.0    
8813                                                                    9.0    
8814                                                                    7.0    

Nombre Servicio de Salud  Metropolitano Sur                                    
0                                                                       NaN 

In [55]:
df_combinado3.groupby('Nombre Servicio de Salud')["Num_Pacientes"].median().reset_index(name='MEdiana')

Nombre Servicio de Salud  MEdiana
0  Metropolitano Central                         ...      9.0
1  Metropolitano Norte                           ...     11.0
2  Metropolitano Occidente                       ...      7.0
3  Metropolitano Oriente                         ...      9.0
4  Metropolitano Sur                             ...      5.0
5  Metropolitano Sur Oriente                     ...      5.0

## Trunca 

# Pregunta 4

In [56]:
Info_sheet

Número de llegada                              Hospital al que llega   
0                      1       Instituto Traumatológico Dr. Teodoro Gebauer  \
1                      2                Hospital Clínico San Borja-Arriarán   
2                      3                         Hospital Dr. Luis Tisné B.   
3                      4              Hospital Dr. Exequiel González Cortés   
4                      5                       Hospital Barros Luco Trudeau   
...                  ...                                                ...   
65529              65530                Hospital Clínico San Borja-Arriarán   
65530              65531              Hospital Dr. Exequiel González Cortés   
65531              65532                              Hospital Del Salvador   
65532              65533  Hospital de Enfermedades Infecciosas Dr. Lucio...   
65533              65534  Hospital de Urgencia Asistencia Pública Dr. Al...   

         GRD  Día de llegada  Día de salida  Estadia  
0       GRD3               1              4        3  
1       GRD7               1              5        4  
2       GRD7               1              5        4  
3      GRD10               1             11       10  
4       GRD6               1              6        5  
...      ...             ...            ...      ...  
65529   GRD2             424            428        4  
65530   GRD9             424            429        5  
65531  GRD10             424            434       10  
65532   GRD3             424            427        3  
65533   GRD7             424            428        4  

[65534 rows x 6 columns]

In [57]:
import pandas as pd

valor_busqueda = 415
columna = 'Día de llegada'

indice_primera_aparicion = Info_sheet.index[Info_sheet[columna] == valor_busqueda].min()

print("El índice de la primera aparición es:", indice_primera_aparicion)


El índice de la primera aparición es: 63989


In [59]:
df4 = Info_sheet.iloc[63989:] #Recorto solo los ultimos 10 dias
result4 = df4.groupby(['Hospital al que llega','GRD',"Día de llegada","Estadia"]).size().reset_index(name='Num_Pacientes') 
result4

Hospital al que llega   GRD  Día de llegada   
0       Complejo Hospitalario Dr. Sótero del Río    GRD1             418  \
1       Complejo Hospitalario Dr. Sótero del Río    GRD1             419   
2       Complejo Hospitalario Dr. Sótero del Río    GRD1             420   
3       Complejo Hospitalario Dr. Sótero del Río    GRD1             422   
4       Complejo Hospitalario Dr. Sótero del Río    GRD1             423   
...                                            ...   ...             ...   
1027  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD9             418   
1028  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD9             420   
1029  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD9             422   
1030  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD9             423   
1031  Instituto de Neurocirugía Dr. Alfonso Asenjo  GRD9             424   

      Estadia  Num_Pacientes  
0          10              1  
1          10              1  
2          10              1  
3          10              2  
4          10              2  
...       ...            ...  
1027        5              2  
1028        5              1  
1029        5              1  
1030        5              1  
1031        5              2  

[1032 rows x 5 columns]

### 8,88

# Pregunta 5
### ¿Cuál cifra corresponde al costo total de gastos, realizados en el hospital del Carmen, para la atención de los pacientes que llegaron los últimos 10 días (desde el día 415 al día 424)?  



In [60]:
Hops_4= Hosp_sheet[7:8].transpose()
Hops_GRD = Hops_4[8:]
Hops_GRD = Hops_GRD.reset_index(drop=False)
nombres_columnas = ['GRD', 'Precio']
Hops_GRD.columns = nombres_columnas
Hops_GRD["GRD"] = ["GRD1","GRD2","GRD3", "GRD4","GRD5","GRD6","GRD7","GRD8","GRD9","GRD10"]
Hops_GRD

GRD Precio
0   GRD1  0.444
1   GRD2  0.439
2   GRD3  0.386
3   GRD4  0.569
4   GRD5  0.404
5   GRD6  0.416
6   GRD7  0.302
7   GRD8  0.414
8   GRD9  0.494
9  GRD10  0.425

In [65]:
Hops_GRD['Precio'].mean()

0.4293

In [61]:
valor_busqueda = "Hospital el Carmen"
columna = 'Hospital al que llega'

indice_primera_aparicion = result4.index[result4[columna] == valor_busqueda].min()

print("El índice de la primera aparición es:", indice_primera_aparicion)


El índice de la primera aparición es: 839


In [80]:
result5 =result4.iloc[839:868]
result5.head()

Hospital al que llega    GRD  Día de llegada  Estadia  Num_Pacientes
839    Hospital el Carmen   GRD1             419       10              1
840    Hospital el Carmen   GRD1             423       10              1
841    Hospital el Carmen  GRD10             416       10              1
842    Hospital el Carmen  GRD10             419       10              1
843    Hospital el Carmen   GRD2             417        4              1

In [81]:
df_combinado5 = pd.merge(result5, Hops_GRD, on='GRD') #combino ambos df por la columna comun
df_combinado5.head()

Hospital al que llega    GRD  Día de llegada  Estadia  Num_Pacientes Precio
0    Hospital el Carmen   GRD1             419       10              1  0.444
1    Hospital el Carmen   GRD1             423       10              1  0.444
2    Hospital el Carmen  GRD10             416       10              1  0.425
3    Hospital el Carmen  GRD10             419       10              1  0.425
4    Hospital el Carmen   GRD2             417        4              1  0.439

In [64]:

df_combinado5["Costo Total"] = df_combinado5["Num_Pacientes"]* df_combinado5["Precio"]*df_combinado5["Estadia"]
df_combinado5["Costo Total"].sum()

77.776

### 77.776

# Pregunta 6
### ¿Cuál es el costo total de gastos del servicio de salud Metropolitano Central, para la atención de los pacientes, que llegaron los últimos 10 días utilizando los costos diarios de atención por GRD promedio del sistema (desde el día 415 al día 424)?  



### Hay que calcular los costos diarios por GRD de todo el sistema

In [146]:
df4 # ULtimos 10 dias
df4 = df4.rename(columns={'Hospital al que llega' : 'Nombre Establecimiento'})
df_combinado6 = pd.merge(df4, HxS, on='Nombre Establecimiento') #combino ambos df por la columna comun
df_combinado6

Número de llegada                      Nombre Establecimiento    GRD   
0                 63990  Complejo Hospitalario Dr. Sótero del Río     GRD2  \
1                 63995  Complejo Hospitalario Dr. Sótero del Río     GRD8   
2                 64006  Complejo Hospitalario Dr. Sótero del Río     GRD2   
3                 64015  Complejo Hospitalario Dr. Sótero del Río     GRD9   
4                 64032  Complejo Hospitalario Dr. Sótero del Río     GRD3   
...                 ...                                         ...    ...   
1540              65375       Hospital Metropolitano ( ex Militar )   GRD6   
1541              65446       Hospital Metropolitano ( ex Militar )   GRD1   
1542              65447       Hospital Metropolitano ( ex Militar )  GRD10   
1543              65461       Hospital Metropolitano ( ex Militar )   GRD5   
1544              65517       Hospital Metropolitano ( ex Militar )   GRD2   

      Día de llegada  Día de salida  Estadia   
0                415            419        4  \
1                415            419        4   
2                415            419        4   
3                415            420        5   
4                415            418        3   
...              ...            ...      ...   
1540             423            428        5   
1541             423            433       10   
1542             423            433       10   
1543             424            428        4   
1544             424            428        4   

                               Nombre Servicio de Salud  
0     Metropolitano Sur Oriente                     ...  
1     Metropolitano Sur Oriente                     ...  
2     Metropolitano Sur Oriente                     ...  
3     Metropolitano Sur Oriente                     ...  
4     Metropolitano Sur Oriente                     ...  
...                                                 ...  
1540  Metropolitano Sur Oriente                     ...  
1541  Metropolitano Sur Oriente                     ...  
1542  Metropolitano Sur Oriente                     ...  
1543  Metropolitano Sur Oriente                     ...  
1544  Metropolitano Sur Oriente                     ...  

[1545 rows x 7 columns]

In [145]:
result6 = df_combinado6.groupby(['Nombre Servicio de Salud','GRD',"Estadia"]).size().reset_index(name='Num_Pacientes') 
Central = result6[:10]

Nombre Servicio de Salud    GRD  Estadia   
0  Metropolitano Central                         ...   GRD1       10  \
1  Metropolitano Central                         ...  GRD10       10   
2  Metropolitano Central                         ...   GRD2        4   
3  Metropolitano Central                         ...   GRD3        3   
4  Metropolitano Central                         ...   GRD4        6   
5  Metropolitano Central                         ...   GRD5        4   
6  Metropolitano Central                         ...   GRD6        5   
7  Metropolitano Central                         ...   GRD7        4   
8  Metropolitano Central                         ...   GRD8        4   
9  Metropolitano Central                         ...   GRD9        5   

   Num_Pacientes  
0              1  
1              1  
2              1  
3              1  
4              1  
5              1  
6              1  
7              1  
8              1  
9              1

In [126]:
GRDs = Hosp_sheet.iloc[:,8:].columns.tolist()
print(GRDs)

['Costo diario GRD1 [MM$]', 'Costo diario GRD2 [MM$]', 'Costo diario GRD3 [MM$]', 'Costo diario GRD4 [MM$]', 'Costo diario GRD5 [MM$]', 'Costo diario GRD6 [MM$]', 'Costo diario GRD7 [MM$]', 'Costo diario GRD8 [MM$]', 'Costo diario GRD9 [MM$]', 'Costo diario GRD10 [MM$]']


In [136]:
Promedios_xGRD = Hosp_sheet.groupby('Nombre Servicio de Salud')[GRDs].mean()
Grds_central = Promedios_xGRD[0:1].transpose() #selecciono solo los del Central
Grds_central = Grds_central.reset_index(drop=False)
Grds_central #Tabla de constes promedios de central por GRD

Nombre Servicio de Salud                     index   
0                          Costo diario GRD1 [MM$]  \
1                          Costo diario GRD2 [MM$]   
2                          Costo diario GRD3 [MM$]   
3                          Costo diario GRD4 [MM$]   
4                          Costo diario GRD5 [MM$]   
5                          Costo diario GRD6 [MM$]   
6                          Costo diario GRD7 [MM$]   
7                          Costo diario GRD8 [MM$]   
8                          Costo diario GRD9 [MM$]   
9                         Costo diario GRD10 [MM$]   

Nombre Servicio de Salud  Metropolitano Central                               
0                                                                  0.473667   
1                                                                  0.437000   
2                                                                  0.434667   
3                                                                  0.498000   
4                                                                  0.387000   
5                                                                  0.462333   
6                                                                  0.445000   
7                                                                  0.435667   
8                                                                  0.381333   
9                                                                  0.522333

In [138]:
nombres_columnas = ['GRD', 'Precio']
Grds_central.columns = nombres_columnas
Grds_central["GRD"] = ["GRD1","GRD2","GRD3", "GRD4","GRD5","GRD6","GRD7","GRD8","GRD9","GRD10"]
Grds_central

GRD    Precio
0   GRD1  0.473667
1   GRD2  0.437000
2   GRD3  0.434667
3   GRD4  0.498000
4   GRD5  0.387000
5   GRD6  0.462333
6   GRD7  0.445000
7   GRD8  0.435667
8   GRD9  0.381333
9  GRD10  0.522333

In [139]:
df_combinado6 = pd.merge(Grds_central, Central, on='GRD') #combino ambos df por la columna comun
df_combinado6

GRD    Precio                           Nombre Servicio de Salud   
0   GRD1  0.473667  Metropolitano Central                         ...  \
1   GRD2  0.437000  Metropolitano Central                         ...   
2   GRD3  0.434667  Metropolitano Central                         ...   
3   GRD4  0.498000  Metropolitano Central                         ...   
4   GRD5  0.387000  Metropolitano Central                         ...   
5   GRD6  0.462333  Metropolitano Central                         ...   
6   GRD7  0.445000  Metropolitano Central                         ...   
7   GRD8  0.435667  Metropolitano Central                         ...   
8   GRD9  0.381333  Metropolitano Central                         ...   
9  GRD10  0.522333  Metropolitano Central                         ...   

   Estadia  Num_Pacientes  
0       10             26  
1        4             30  
2        3             25  
3        6             29  
4        4             20  
5        5             25  
6        4             32  
7        4             28  
8        5             25  
9       10             28

In [144]:
df_combinado6["Costo Total"] = df_combinado6["Num_Pacientes"]* df_combinado6["Precio"]*df_combinado6["Estadia"]
df_combinado6['Costo Total'].sum()

683.2716666666666

# Pregunta7
### ¿Qué servicio de salud gastó más dinero, para la atención de los pacientes que llegaron los últimos 10 días utilizando los costos diarios de atención por GRD promedio de cada sistema (desde el día 415 al día 424)?  
#### Se vuelve generica la 6


In [149]:
df_combinado6

Número de llegada                      Nombre Establecimiento    GRD   
0                 63990  Complejo Hospitalario Dr. Sótero del Río     GRD2  \
1                 63995  Complejo Hospitalario Dr. Sótero del Río     GRD8   
2                 64006  Complejo Hospitalario Dr. Sótero del Río     GRD2   
3                 64015  Complejo Hospitalario Dr. Sótero del Río     GRD9   
4                 64032  Complejo Hospitalario Dr. Sótero del Río     GRD3   
...                 ...                                         ...    ...   
1540              65375       Hospital Metropolitano ( ex Militar )   GRD6   
1541              65446       Hospital Metropolitano ( ex Militar )   GRD1   
1542              65447       Hospital Metropolitano ( ex Militar )  GRD10   
1543              65461       Hospital Metropolitano ( ex Militar )   GRD5   
1544              65517       Hospital Metropolitano ( ex Militar )   GRD2   

      Día de llegada  Día de salida  Estadia   
0                415            419        4  \
1                415            419        4   
2                415            419        4   
3                415            420        5   
4                415            418        3   
...              ...            ...      ...   
1540             423            428        5   
1541             423            433       10   
1542             423            433       10   
1543             424            428        4   
1544             424            428        4   

                               Nombre Servicio de Salud  
0     Metropolitano Sur Oriente                     ...  
1     Metropolitano Sur Oriente                     ...  
2     Metropolitano Sur Oriente                     ...  
3     Metropolitano Sur Oriente                     ...  
4     Metropolitano Sur Oriente                     ...  
...                                                 ...  
1540  Metropolitano Sur Oriente                     ...  
1541  Metropolitano Sur Oriente                     ...  
1542  Metropolitano Sur Oriente                     ...  
1543  Metropolitano Sur Oriente                     ...  
1544  Metropolitano Sur Oriente                     ...  

[1545 rows x 7 columns]

In [177]:
result7 = df_combinado6.groupby(['Nombre Servicio de Salud','GRD',"Estadia"]).size().reset_index(name='Num_Pacientes')
result7 #para mergear con la columna de costos medios de GRD

Nombre Servicio de Salud    GRD  Estadia   
0   Metropolitano Central                         ...   GRD1       10  \
1   Metropolitano Central                         ...  GRD10       10   
2   Metropolitano Central                         ...   GRD2        4   
3   Metropolitano Central                         ...   GRD3        3   
4   Metropolitano Central                         ...   GRD4        6   
5   Metropolitano Central                         ...   GRD5        4   
6   Metropolitano Central                         ...   GRD6        5   
7   Metropolitano Central                         ...   GRD7        4   
8   Metropolitano Central                         ...   GRD8        4   
9   Metropolitano Central                         ...   GRD9        5   
10  Metropolitano Norte                           ...   GRD1       10   
11  Metropolitano Norte                           ...  GRD10       10   
12  Metropolitano Norte                           ...   GRD2        4   
13  Metropolitano Norte                           ...   GRD3        3   
14  Metropolitano Norte                           ...   GRD4        6   
15  Metropolitano Norte                           ...   GRD5        4   
16  Metropolitano Norte                           ...   GRD6        5   
17  Metropolitano Norte                           ...   GRD7        4   
18  Metropolitano Norte                           ...   GRD8        4   
19  Metropolitano Norte                           ...   GRD9        5   
20  Metropolitano Occidente                       ...   GRD1       10   
21  Metropolitano Occidente                       ...  GRD10       10   
22  Metropolitano Occidente                       ...   GRD2        4   
23  Metropolitano Occidente                       ...   GRD3        3   
24  Metropolitano Occidente                       ...   GRD4        6   
25  Metropolitano Occidente                       ...   GRD5        4   
26  Metropolitano Occidente                       ...   GRD6        5   
27  Metropolitano Occidente                       ...   GRD7        4   
28  Metropolitano Occidente                       ...   GRD8        4   
29  Metropolitano Occidente                       ...   GRD9        5   
30  Metropolitano Oriente                         ...   GRD1       10   
31  Metropolitano Oriente                         ...  GRD10       10   
32  Metropolitano Oriente                         ...   GRD2        4   
33  Metropolitano Oriente                         ...   GRD3        3   
34  Metropolitano Oriente                         ...   GRD4        6   
35  Metropolitano Oriente                         ...   GRD5        4   
36  Metropolitano Oriente                         ...   GRD6        5   
37  Metropolitano Oriente                         ...   GRD7        4   
38  Metropolitano Oriente                         ...   GRD8        4   
39  Metropolitano Oriente                         ...   GRD9        5   
40  Metropolitano Sur                             ...   GRD1       10   
41  Metropolitano Sur                             ...  GRD10       10   
42  Metropolitano Sur                             ...   GRD2        4   
43  Metropolitano Sur                             ...   GRD3        3   
44  Metropolitano Sur                             ...   GRD4        6   
45  Metropolitano Sur                             ...   GRD5        4   
46  Metropolitano Sur                             ...   GRD6        5   
47  Metropolitano Sur                             ...   GRD7        4   
48  Metropolitano Sur                             ...   GRD8        4   
49  Metropolitano Sur                             ...   GRD9        5   
50  Metropolitano Sur Oriente                     ...   GRD1       10   
51  Metropolitano Sur Oriente                     ...  GRD10       10   
52  Metropolitano Sur Oriente                     ...   GRD2        4   
53  Metropolitano Sur Oriente                     ...   GRD3        3   
54  Metropolit

In [241]:
result7['Etiqueta'] = result7['GRD'] + '-' + result7['Nombre Servicio de Salud']
result7_1 = result7.iloc[:,2:-1]
nombres_columnas = ['Estadia', 'Num_Pacientes','Etiqueta']
result7_1.columns = nombres_columnas
result7_1

Estadia  Num_Pacientes                                           Etiqueta
0        10             26  GRD1-Metropolitano Central                    ...
1        10             28  GRD10-Metropolitano Central                   ...
2         4             30  GRD2-Metropolitano Central                    ...
3         3             25  GRD3-Metropolitano Central                    ...
4         6             29  GRD4-Metropolitano Central                    ...
5         4             20  GRD5-Metropolitano Central                    ...
6         5             25  GRD6-Metropolitano Central                    ...
7         4             32  GRD7-Metropolitano Central                    ...
8         4             28  GRD8-Metropolitano Central                    ...
9         5             25  GRD9-Metropolitano Central                    ...
10       10             12  GRD1-Metropolitano Norte                      ...
11       10             16  GRD10-Metropolitano Norte                     ...
12        4             10  GRD2-Metropolitano Norte                      ...
13        3             12  GRD3-Metropolitano Norte                      ...
14        6              8  GRD4-Metropolitano Norte                      ...
15        4              8  GRD5-Metropolitano Norte                      ...
16        5             12  GRD6-Metropolitano Norte                      ...
17        4             13  GRD7-Metropolitano Norte                      ...
18        4              8  GRD8-Metropolitano Norte                      ...
19        5              7  GRD9-Metropolitano Norte                      ...
20       10             32  GRD1-Metropolitano Occidente                  ...
21       10             20  GRD10-Metropolitano Occidente                 ...
22        4             26  GRD2-Metropolitano Occidente                  ...
23        3             33  GRD3-Metropolitano Occidente                  ...
24        6             20  GRD4-Metropolitano Occidente                  ...
25        4             28  GRD5-Metropolitano Occidente                  ...
26        5             25  GRD6-Metropolitano Occidente                  ...
27        4             34  GRD7-Metropolitano Occidente                  ...
28        4             22  GRD8-Metropolitano Occidente                  ...
29        5             17  GRD9-Metropolitano Occidente                  ...
30       10             32  GRD1-Metropolitano Oriente                    ...
31       10             41  GRD10-Metropolitano Oriente                   ...
32        4             39  GRD2-Metropolitano Oriente                    ...
33        3             36  GRD3-Metropolitano Oriente                    ...
34        6             31  GRD4-Metropolitano Oriente                    ...
35        4             35  GRD5-Metropolitano Oriente                    ...
36        5             42  GRD6-Metropolitano Oriente                    ...
37        4             44  GRD7-Metropolitano Oriente                    ...
38        4             39  GRD8-Metropolitano Oriente                    ...
39        5             37  GRD9-Metropolitano Oriente                    ...
40       10             23  GRD1-Metropolitano Sur                        ...
41       10             33  GRD10-Metropolitano Sur                       ...
42        4             35  GRD2-Metropolitano Sur                        ...
43        3             37  GRD3-Metropolitano Sur                        ...
44        6             31  GRD4-Metropolitano Sur                        ...
45        4             34  GRD5-Metropolitano Sur                        ...
46        5             33  GRD6-Metropolitano Sur                        ...
47        4             26  GRD7-Metropolitano Sur                        ...
48        4             31  GRD8-Metropolitano Sur                        ...
49        5             40  GRD9-Metropolitano Sur                        ...
50       10             26

In [200]:
GRD7 = Hosp_sheet.groupby('Nombre Servicio de Salud')[GRDs].mean()
GRD7.transpose()
GRD7_index = GRD7.transpose().reset_index(drop=False)
nuevos_nombres = ['GRD'] + list(GRD7_index.columns[1:])
GRD7_index.columns = nuevos_nombres
GRD7_index

GRD   
0   Costo diario GRD1 [MM$]  \
1   Costo diario GRD2 [MM$]   
2   Costo diario GRD3 [MM$]   
3   Costo diario GRD4 [MM$]   
4   Costo diario GRD5 [MM$]   
5   Costo diario GRD6 [MM$]   
6   Costo diario GRD7 [MM$]   
7   Costo diario GRD8 [MM$]   
8   Costo diario GRD9 [MM$]   
9  Costo diario GRD10 [MM$]   

   Metropolitano Central                                
0                                           0.473667   \
1                                           0.437000    
2                                           0.434667    
3                                           0.498000    
4                                           0.387000    
5                                           0.462333    
6                                           0.445000    
7                                           0.435667    
8                                           0.381333    
9                                           0.522333    

   Metropolitano Norte                                  
0                                              0.593   \
1                                              0.509    
2                                              0.367    
3                                              0.492    
4                                              0.524    
5                                              0.405    
6                                              0.571    
7                                              0.474    
8                                              0.498    
9                                              0.305    

   Metropolitano Occidente                              
0                                            0.49050   \
1                                            0.42125    
2                                            0.45900    
3                                            0.38875    
4                                            0.51025    
5                                            0.41225    
6                                            0.45600    
7                                            0.36925    
8                                            0.48975    
9                                            0.36550    

   Metropolitano Oriente                                
0                                            0.42600   \
1                                            0.44050    
2                                            0.44850    
3                                            0.44375    
4                                            0.52000    
5                                            0.39625    
6                                            0.40300    
7                                            0.47050    
8                                            0.42350    
9                                            0.43075    

   Metropolitano Sur                                    
0                                             0.5152   \
1                                             0.4468    
2                                             0.5428    
3                                             0.4390    
4                                             0.5190    
5                                             0.4038    
6                                             0.3892    
7                                             0.5152    
8                                             0.4890    
9                                             0.3836    

   Metropolitano Sur Oriente                           
0                                            0.52875   
1                                            0.51800   
2                                            0.46825   
3                                            0.47675   
4                                            0.48375   
5                                            0.44575   
6                                            0.48900   
7                                            0.45725   
8                                     

In [201]:
GRD7_index["GRD"] = ["GRD1","GRD2","GRD3", "GRD4","GRD5","GRD6","GRD7","GRD8","GRD9","GRD10"]
GRD7_index

GRD  Metropolitano Central                                
0   GRD1                                           0.473667   \
1   GRD2                                           0.437000    
2   GRD3                                           0.434667    
3   GRD4                                           0.498000    
4   GRD5                                           0.387000    
5   GRD6                                           0.462333    
6   GRD7                                           0.445000    
7   GRD8                                           0.435667    
8   GRD9                                           0.381333    
9  GRD10                                           0.522333    

   Metropolitano Norte                                  
0                                              0.593   \
1                                              0.509    
2                                              0.367    
3                                              0.492    
4                                              0.524    
5                                              0.405    
6                                              0.571    
7                                              0.474    
8                                              0.498    
9                                              0.305    

   Metropolitano Occidente                              
0                                            0.49050   \
1                                            0.42125    
2                                            0.45900    
3                                            0.38875    
4                                            0.51025    
5                                            0.41225    
6                                            0.45600    
7                                            0.36925    
8                                            0.48975    
9                                            0.36550    

   Metropolitano Oriente                                
0                                            0.42600   \
1                                            0.44050    
2                                            0.44850    
3                                            0.44375    
4                                            0.52000    
5                                            0.39625    
6                                            0.40300    
7                                            0.47050    
8                                            0.42350    
9                                            0.43075    

   Metropolitano Sur                                    
0                                             0.5152   \
1                                             0.4468    
2                                             0.5428    
3                                             0.4390    
4                                             0.5190    
5                                             0.4038    
6                                             0.3892    
7                                             0.5152    
8                                             0.4890    
9                                             0.3836    

   Metropolitano Sur Oriente                           
0                                            0.52875   
1                                            0.51800   
2                                            0.46825   
3                                            0.47675   
4                                            0.48375   
5                                            0.44575   
6                                            0.48900   
7                                            0.45725   
8                                            0.47025   
9                                            0.47375

In [202]:
GRD7_index

GRD  Metropolitano Central                                
0   GRD1                                           0.473667   \
1   GRD2                                           0.437000    
2   GRD3                                           0.434667    
3   GRD4                                           0.498000    
4   GRD5                                           0.387000    
5   GRD6                                           0.462333    
6   GRD7                                           0.445000    
7   GRD8                                           0.435667    
8   GRD9                                           0.381333    
9  GRD10                                           0.522333    

   Metropolitano Norte                                  
0                                              0.593   \
1                                              0.509    
2                                              0.367    
3                                              0.492    
4                                              0.524    
5                                              0.405    
6                                              0.571    
7                                              0.474    
8                                              0.498    
9                                              0.305    

   Metropolitano Occidente                              
0                                            0.49050   \
1                                            0.42125    
2                                            0.45900    
3                                            0.38875    
4                                            0.51025    
5                                            0.41225    
6                                            0.45600    
7                                            0.36925    
8                                            0.48975    
9                                            0.36550    

   Metropolitano Oriente                                
0                                            0.42600   \
1                                            0.44050    
2                                            0.44850    
3                                            0.44375    
4                                            0.52000    
5                                            0.39625    
6                                            0.40300    
7                                            0.47050    
8                                            0.42350    
9                                            0.43075    

   Metropolitano Sur                                    
0                                             0.5152   \
1                                             0.4468    
2                                             0.5428    
3                                             0.4390    
4                                             0.5190    
5                                             0.4038    
6                                             0.3892    
7                                             0.5152    
8                                             0.4890    
9                                             0.3836    

   Metropolitano Sur Oriente                           
0                                            0.52875   
1                                            0.51800   
2                                            0.46825   
3                                            0.47675   
4                                            0.48375   
5                                            0.44575   
6                                            0.48900   
7                                            0.45725   
8                                            0.47025   
9                                            0.47375

In [233]:
# Obtener las etiquetas de las columnas (excepto la primera)
columnas = GRD7_index.columns[1:]
# Crear una lista para almacenar los datos del DataFrame resultante
datos_resultantes = []

# Iterar sobre las columnas y concatenar la primera columna original con el nombre de cada columna
for columna in columnas:
    etiqueta = GRD7_index['GRD'].astype(str) + '-' + columna
    valores = GRD7_index[columna].tolist()
    datos_resultantes.extend(list(zip(etiqueta, valores)))

# Crear el DataFrame resultante
df_resultante = pd.DataFrame(datos_resultantes, columns=['Etiqueta', 'Valor'])

# Mostrar el DataFrame resultante de 60x2
df_resultante

Index(['Metropolitano Central                             ',
       'Metropolitano Norte                               ',
       'Metropolitano Occidente                           ',
       'Metropolitano Oriente                             ',
       'Metropolitano Sur                                 ',
       'Metropolitano Sur Oriente                         '],
      dtype='object')


Etiqueta     Valor
0   GRD1-Metropolitano Central                    ...  0.473667
1   GRD2-Metropolitano Central                    ...  0.437000
2   GRD3-Metropolitano Central                    ...  0.434667
3   GRD4-Metropolitano Central                    ...  0.498000
4   GRD5-Metropolitano Central                    ...  0.387000
5   GRD6-Metropolitano Central                    ...  0.462333
6   GRD7-Metropolitano Central                    ...  0.445000
7   GRD8-Metropolitano Central                    ...  0.435667
8   GRD9-Metropolitano Central                    ...  0.381333
9   GRD10-Metropolitano Central                   ...  0.522333
10  GRD1-Metropolitano Norte                      ...  0.593000
11  GRD2-Metropolitano Norte                      ...  0.509000
12  GRD3-Metropolitano Norte                      ...  0.367000
13  GRD4-Metropolitano Norte                      ...  0.492000
14  GRD5-Metropolitano Norte                      ...  0.524000
15  GRD6-Metropolitano Norte                      ...  0.405000
16  GRD7-Metropolitano Norte                      ...  0.571000
17  GRD8-Metropolitano Norte                      ...  0.474000
18  GRD9-Metropolitano Norte                      ...  0.498000
19  GRD10-Metropolitano Norte                     ...  0.305000
20  GRD1-Metropolitano Occidente                  ...  0.490500
21  GRD2-Metropolitano Occidente                  ...  0.421250
22  GRD3-Metropolitano Occidente                  ...  0.459000
23  GRD4-Metropolitano Occidente                  ...  0.388750
24  GRD5-Metropolitano Occidente                  ...  0.510250
25  GRD6-Metropolitano Occidente                  ...  0.412250
26  GRD7-Metropolitano Occidente                  ...  0.456000
27  GRD8-Metropolitano Occidente                  ...  0.369250
28  GRD9-Metropolitano Occidente                  ...  0.489750
29  GRD10-Metropolitano Occidente                 ...  0.365500
30  GRD1-Metropolitano Oriente                    ...  0.426000
31  GRD2-Metropolitano Oriente                    ...  0.440500
32  GRD3-Metropolitano Oriente                    ...  0.448500
33  GRD4-Metropolitano Oriente                    ...  0.443750
34  GRD5-Metropolitano Oriente                    ...  0.520000
35  GRD6-Metropolitano Oriente                    ...  0.396250
36  GRD7-Metropolitano Oriente                    ...  0.403000
37  GRD8-Metropolitano Oriente                    ...  0.470500
38  GRD9-Metropolitano Oriente                    ...  0.423500
39  GRD10-Metropolitano Oriente                   ...  0.430750
40  GRD1-Metropolitano Sur                        ...  0.515200
41  GRD2-Metropolitano Sur                        ...  0.446800
42  GRD3-Metropolitano Sur                        ...  0.542800
43  GRD4-Metropolitano Sur                        ...  0.439000
44  GRD5-Metropolitano Sur                        ...  0.519000
45  GRD6-Metropolitano Sur                        ...  0.403800
46  GRD7-Metropolitano Sur                        ...  0.389200
47  GRD8-Metropolitano Sur                        ...  0.515200
48  GRD9-Metropolitano Sur                        ...  0.489000
49  GRD10-Metropolitano Sur                       ...  0.383600
50  GRD1-Metropolitano Sur Oriente                ...  0.528750
51  GRD2-Metropolitano Sur Oriente                ...  0.518000
52  GRD3-Metropolitano Sur Oriente                ...  0.468250
53  GRD4-Metropolitano Sur Oriente                ...  0.476750
54  GRD5-Metropolitano Sur Oriente                ...  0.483750
55  GRD6-Metropolitano Sur Oriente                ...  0.445750
56  GRD7-Metropolitano Sur Oriente                ...  0.489000
57  GRD8-Metropolitano Sur Oriente                ...  0.457250
58  GRD9-Metropolitano Sur Oriente                ...  0.470250
59  GRD10-Metropolitano Sur Oriente               ...  0.473750

In [251]:
Reult7_F = pd.merge(df_resultante, result7_1, on='Etiqueta') #combino ambos df por la columna comun
Reult7_F['Costo'] = Reult7_F['Valor'] * Reult7_F['Estadia'] * Reult7_F['Num_Pacientes']
Reult7_F

Etiqueta     Valor  Estadia   
0   GRD1-Metropolitano Central                    ...  0.473667       10  \
1   GRD2-Metropolitano Central                    ...  0.437000        4   
2   GRD3-Metropolitano Central                    ...  0.434667        3   
3   GRD4-Metropolitano Central                    ...  0.498000        6   
4   GRD5-Metropolitano Central                    ...  0.387000        4   
5   GRD6-Metropolitano Central                    ...  0.462333        5   
6   GRD7-Metropolitano Central                    ...  0.445000        4   
7   GRD8-Metropolitano Central                    ...  0.435667        4   
8   GRD9-Metropolitano Central                    ...  0.381333        5   
9   GRD10-Metropolitano Central                   ...  0.522333       10   
10  GRD1-Metropolitano Norte                      ...  0.593000       10   
11  GRD2-Metropolitano Norte                      ...  0.509000        4   
12  GRD3-Metropolitano Norte                      ...  0.367000        3   
13  GRD4-Metropolitano Norte                      ...  0.492000        6   
14  GRD5-Metropolitano Norte                      ...  0.524000        4   
15  GRD6-Metropolitano Norte                      ...  0.405000        5   
16  GRD7-Metropolitano Norte                      ...  0.571000        4   
17  GRD8-Metropolitano Norte                      ...  0.474000        4   
18  GRD9-Metropolitano Norte                      ...  0.498000        5   
19  GRD10-Metropolitano Norte                     ...  0.305000       10   
20  GRD1-Metropolitano Occidente                  ...  0.490500       10   
21  GRD2-Metropolitano Occidente                  ...  0.421250        4   
22  GRD3-Metropolitano Occidente                  ...  0.459000        3   
23  GRD4-Metropolitano Occidente                  ...  0.388750        6   
24  GRD5-Metropolitano Occidente                  ...  0.510250        4   
25  GRD6-Metropolitano Occidente                  ...  0.412250        5   
26  GRD7-Metropolitano Occidente                  ...  0.456000        4   
27  GRD8-Metropolitano Occidente                  ...  0.369250        4   
28  GRD9-Metropolitano Occidente                  ...  0.489750        5   
29  GRD10-Metropolitano Occidente                 ...  0.365500       10   
30  GRD1-Metropolitano Oriente                    ...  0.426000       10   
31  GRD2-Metropolitano Oriente                    ...  0.440500        4   
32  GRD3-Metropolitano Oriente                    ...  0.448500        3   
33  GRD4-Metropolitano Oriente                    ...  0.443750        6   
34  GRD5-Metropolitano Oriente                    ...  0.520000        4   
35  GRD6-Metropolitano Oriente                    ...  0.396250        5   
36  GRD7-Metropolitano Oriente                    ...  0.403000        4   
37  GRD8-Metropolitano Oriente                    ...  0.470500        4   
38  GRD9-Metropolitano Oriente                    ...  0.423500        5   
39  GRD10-Metropolitano Oriente                   ...  0.430750       10   
40  GRD1-Metropolitano Sur                        ...  0.515200       10   
41  GRD2-Metropolitano Sur                        ...  0.446800        4   
42  GRD3-Metropolitano Sur                        ...  0.542800        3   
43  GRD4-Metropolitano Sur                        ...  0.439000        6   
44  GRD5-Metropolitano Sur                        ...  0.519000        4   
45  GRD6-Metropolitano Sur                        ...  0.403800        5   
46  GRD7-Metropolitano Sur                        ...  0.389200        4   
47  GRD8-Metropolitano Sur                        ...  0.515200        4   
48  GRD9-Metropolitano Sur                        ...  0.489000        5   
49  GRD10-Metropolitano Sur                       ...  0.383600       10   
50  GRD1-Metropolitano Sur Oriente                ...  0.528750       10   
51  GRD2-Metropolitano Sur Oriente                ...  0.518000        4   
52  GRD3-Metropol

In [254]:

# Definir el tamaño del intervalo
intervalo = 10

# Calcular la suma en intervalos
suma_intervalos = []
for i in range(0, len(Reult7_F), intervalo):
    suma = Reult7_F.loc[i:i+intervalo-1, 'Costo'].sum()
    suma_intervalos.append(suma)

# Mostrar los resultados
suma_intervalos

[683.2716666666666,
 280.50600000000003,
 610.779,
 891.307,
 788.9133999999999,
 597.38475]

# Pregunta 8
##### 

### 891.307 - 280.506 = 610,801

# Pregunta 9

### ¿Cuál servicio de salud tuvo el costo de instalación de las 8 camas más baratas, siendo el mínimo?  



In [269]:
Hosp_sheet

Nombre Servicio de Salud   
0   Metropolitano Norte                           ...  \
1   Metropolitano Occidente                       ...   
2   Metropolitano Occidente                       ...   
3   Metropolitano Occidente                       ...   
4   Metropolitano Occidente                       ...   
5   Metropolitano Central                         ...   
6   Metropolitano Central                         ...   
7   Metropolitano Central                         ...   
8   Metropolitano Oriente                         ...   
9   Metropolitano Oriente                         ...   
10  Metropolitano Oriente                         ...   
11  Metropolitano Oriente                         ...   
12  Metropolitano Sur                             ...   
13  Metropolitano Sur                             ...   
14  Metropolitano Sur                             ...   
15  Metropolitano Sur                             ...   
16  Metropolitano Sur                             ...   
17  Metropolitano Sur Oriente                     ...   
18  Metropolitano Sur Oriente                     ...   
19  Metropolitano Sur Oriente                     ...   
20  Metropolitano Sur Oriente                     ...   

                               Nombre Establecimiento   
0                      Complejo Hospitalario San José  \
1              Hospital Adalberto Steeger (Talagante)   
2                     Hospital Dr. Félix Bulnes Cerda   
3                           Hospital San Juan de Dios   
4        Instituto Traumatológico Dr. Teodoro Gebauer   
5                 Hospital Clínico San Borja-Arriarán   
6   Hospital de Urgencia Asistencia Pública Dr. Al...   
7                                  Hospital el Carmen   
8                               Hospital Del Salvador   
9                          Hospital Dr. Luis Tisné B.   
10       Instituto de Neurocirugía Dr. Alfonso Asenjo   
11  Instituto Nacional de Enfermedades Respiratori...   
12                       Hospital Barros Luco Trudeau   
13  Hospital de Enfermedades Infecciosas Dr. Lucio...   
14              Hospital Dr. Exequiel González Cortés   
15                                   Hospital El Pino   
16          Hospital Parroquial de San Bernardo         
17         Complejo Hospitalario Dr. Sótero del Río     
18                     Hospital Clínico de la Florida   
19              Hospital Metropolitano ( ex Militar )   
20         Hospital Padre Alberto Hurtado (San Ramón)   

    Área Cuidados Intensivos Adultos  Área Cuidados Intermedios Adulto   
0                                 11                                33  \
1                                  6                                11   
2                                  3                                 6   
3                                 29                                47   
4                                 12                                 6   
5                                 17                                26   
6                                 24                                20   
7                                  6                                12   
8                                 22                                24   
9                                  8                                16   
10                                12                                11   
11                                15                                29   
12                                31                                43   
13                                11                                 9   
14                                23                                 8   
15                                12                                20   
16                                 6                                10   
17                                20                                35   
18                                 9                                15   
19                                12   

In [277]:
result9 = Hosp_sheet.groupby(['Nombre Servicio de Salud','Costo de instalar una cama [MM$]'])['Máxima cantidad de camas a aumetar'].sum().reset_index(name='Máxima cantidad de camas a aumetar')

Hosp_sheet_filtrado = Hosp_sheet[Hosp_sheet['Máxima cantidad de camas a aumetar'] > 8]
Hosp_sheet_filtrado


result9.sort_values(by=['Costo de instalar una cama [MM$]'],ascending = False)

Nombre Servicio de Salud   
7   Metropolitano Occidente                       ...  \
11  Metropolitano Oriente                         ...   
3   Metropolitano Norte                           ...   
2   Metropolitano Central                         ...   
1   Metropolitano Central                         ...   
16  Metropolitano Sur                             ...   
6   Metropolitano Occidente                       ...   
15  Metropolitano Sur                             ...   
5   Metropolitano Occidente                       ...   
0   Metropolitano Central                         ...   
10  Metropolitano Oriente                         ...   
20  Metropolitano Sur Oriente                     ...   
14  Metropolitano Sur                             ...   
19  Metropolitano Sur Oriente                     ...   
18  Metropolitano Sur Oriente                     ...   
9   Metropolitano Oriente                         ...   
13  Metropolitano Sur                             ...   
4   Metropolitano Occidente                       ...   
8   Metropolitano Oriente                         ...   
12  Metropolitano Sur                             ...   
17  Metropolitano Sur Oriente                     ...   

    Costo de instalar una cama [MM$]  Máxima cantidad de camas a aumetar  
7                               1433                                   5  
11                              1422                                  15  
3                               1407                                  10  
2                               1368                                   4  
1                               1347                                  10  
16                              1319                                  25  
6                               1310                                   5  
15                              1304                                   5  
5                               1263                                   5  
0                               1256                                  10  
10                              1256                                   8  
20                              1203                                  12  
14                              1193                                   5  
19                              1166                                   5  
18                              1093                                   5  
9                               1075                                  12  
13                              1074                                   9  
4                               1066                                  30  
8                               1056                                   8  
12                              1051                                   4  
17                              1028                                   5

In [260]:
Hosp_sheet_filtrado

Nombre Servicio de Salud   
0   Metropolitano Norte                           ...  \
3   Metropolitano Occidente                       ...   
5   Metropolitano Central                         ...   
6   Metropolitano Central                         ...   
8   Metropolitano Oriente                         ...   
11  Metropolitano Oriente                         ...   
12  Metropolitano Sur                             ...   
15  Metropolitano Sur                             ...   
17  Metropolitano Sur Oriente                     ...   

                               Nombre Establecimiento   
0                      Complejo Hospitalario San José  \
3                           Hospital San Juan de Dios   
5                 Hospital Clínico San Borja-Arriarán   
6   Hospital de Urgencia Asistencia Pública Dr. Al...   
8                               Hospital Del Salvador   
11  Instituto Nacional de Enfermedades Respiratori...   
12                       Hospital Barros Luco Trudeau   
15                                   Hospital El Pino   
17         Complejo Hospitalario Dr. Sótero del Río     

    Área Cuidados Intensivos Adultos  Área Cuidados Intermedios Adulto   
0                                 11                                33  \
3                                 29                                47   
5                                 17                                26   
6                                 24                                20   
8                                 22                                24   
11                                15                                29   
12                                31                                43   
15                                12                                20   
17                                20                                35   

    Total Camas Críticas  Máxima cantidad de camas a aumetar   
0                     44                                  10  \
3                     76                                  30   
5                     43                                  10   
6                     44                                  10   
8                     46                                  12   
11                    44                                  15   
12                    74                                  25   
15                    32                                   9   
17                    55                                  12   

    Costo de instalar una cama [MM$]   
0                               1407  \
3                               1066   
5                               1347   
6                               1256   
8                               1075   
11                              1422   
12                              1319   
15                              1074   
17                              1203   

    Costo de operación anual de una cama [MM$]  Costo diario GRD1 [MM$]   
0                                         47.2                    0.593  \
3                                         39.2                    0.482   
5                                         63.1                    0.509   
6                                         44.5                    0.468   
8                                         45.4                    0.444   
11                                        63.0                    0.498   
12                                        42.9                    0.551   
15                                        66.1                    0.546   
17                                        40.6                    0.560   

    Costo diario GRD2 [MM$]  Costo diario GRD3 [MM$]  Costo diario GRD4 [MM$]   
0                     0.509                    0.367                    0.492  \
3                     0.554                    0.564                    0.331   
5                     0.363                    0.575                    0.534   
6                     0.509  

# Pregunta 10

### ¿Cuál código permite determinar qué servicios de salud tienen más camas intermedias, que camas críticas?  



In [281]:
hospitales = Hosp_sheet.loc[Hosp_sheet["Área Cuidados Intensivos Adultos"] < Hosp_sheet["Área Cuidados Intermedios Adulto"]]

hospitales[["Nombre Establecimiento","Área Cuidados Intensivos Adultos","Área Cuidados Intermedios Adulto"]] 

Nombre Establecimiento   
0                      Complejo Hospitalario San José  \
1              Hospital Adalberto Steeger (Talagante)   
2                     Hospital Dr. Félix Bulnes Cerda   
3                           Hospital San Juan de Dios   
5                 Hospital Clínico San Borja-Arriarán   
7                                  Hospital el Carmen   
8                               Hospital Del Salvador   
9                          Hospital Dr. Luis Tisné B.   
11  Instituto Nacional de Enfermedades Respiratori...   
12                       Hospital Barros Luco Trudeau   
15                                   Hospital El Pino   
16          Hospital Parroquial de San Bernardo         
17         Complejo Hospitalario Dr. Sótero del Río     
18                     Hospital Clínico de la Florida   
19              Hospital Metropolitano ( ex Militar )   
20         Hospital Padre Alberto Hurtado (San Ramón)   

    Área Cuidados Intensivos Adultos  Área Cuidados Intermedios Adulto  
0                                 11                                33  
1                                  6                                11  
2                                  3                                 6  
3                                 29                                47  
5                                 17                                26  
7                                  6                                12  
8                                 22                                24  
9                                  8                                16  
11                                15                                29  
12                                31                                43  
15                                12                                20  
16                                 6                                10  
17                                20                                35  
18                                 9                                15  
19                                12                                20  
20                                 6                                10